apps/src/templates/instructions/codeReview/CommentEditor.jsx

```diff
@@ -6,14 +6,53 @@ import Button from '@cdo/apps/templates/Button';
 import color from '@cdo/apps/util/color';
 
 export default class CommentEditor extends Component {
-  static propTypes = {onNewCommentSubmit: PropTypes.func.isRequired};
+  static propTypes = {
+    onNewCommentSubmit: PropTypes.func.isRequired,
+    onNewCommentCancel: PropTypes.func.isRequired,
+    saveError: PropTypes.bool,
+    saveInProgress: PropTypes.bool
+  };
 
   state = {comment: ''};
 
   commentChanged = event => this.setState({comment: event.target.value});
 
+  renderSaveStatus() {
+    const {saveError, saveInProgress} = this.props;
+    let icon = '';
+    let saveMessageTitle = '';
+    let saveMessageText = '';
+
+    if (saveInProgress) {
+      icon = <span className="fa fa-spin fa-spinner" style={styles.spinner} />;
+      saveMessageTitle = javalabMsg.saving();
+    } else if (saveError) {
+      icon = (
+        <span className="fa fa-exclamation-circle" style={styles.iconError} />
+      );
+      saveMessageTitle = javalabMsg.genericSaveErrorTitle();
+      saveMessageText = javalabMsg.genericErrorMessage();
+    }
+
+    return (
+      <div style={styles.saveStatus}>
+        <div style={styles.saveStatusIcon}>{icon}</div>
+        <div style={styles.saveMessage}>
+          <div style={styles.saveMessageTitle}>{saveMessageTitle}</div>
```

Comment:
Out of curiosity...what happens if we use more descriptive HTML tags for text containers (like a p tag or something)? Do we get unwanted formatting?

Reply:
switched to p tag for the text!

Reply:
to your question, to keep the formatting I had to override the bottom margin to 0 px, because paragraphs add a large bottom margin, but that is pretty easy :)